BerghAdmin/Services/UserManagement/UserService.cs

```diff
@@ -0,0 +1,164 @@
+﻿using Microsoft.AspNetCore.Identity;
+using System.Security.Claims;
+using BerghAdmin.Authorization;
+
+namespace BerghAdmin.Services.UserManagement;
+
+public class UserService : IUserService
+{
+    private readonly UserManager<User> _userManager;
+
+    public UserService(UserManager<User> userManager)
+    {
+        this._userManager = userManager;
+    }
+
+    public Task DeleteUserAsync(string naam)
+    {
+        throw new NotImplementedException();
+    }
+
+    public IList<Claim> GetClaims()
+    {
+        return new List<Claim>()
+        {
+            AdministratorPolicyHandler.Claim,
+            BeheerAmbassadeursPolicyHandler.Claim,
+            BeheerFietsersPolicyHandler.Claim,
+            BeheerFinancienPolicyHandler.Claim,
+            BeheerGolfersPolicyHandler.Claim,
+            BeheerSecretariaatPolicyHandler.Claim
+        };
+    }
+
+    public async Task<User> GetUserAsync(string naam)
+        => await this._userManager.FindByNameAsync(naam);
+
+    public IList<User> GetUsers()
+        =>_userManager.Users.ToList();
+
+    public async Task<IList<Claim>> GetUserClaimsAsync(string naam)
+    {
+        var user = await GetUserAsync(naam);
+        if (user == null)
+            return new List<Claim>();
+
+        var userClaims = await _userManager.GetClaimsAsync(user);
+
+        return userClaims;
+    }
+
+    public async Task<IdentityResult> InsertUserAsync(User user, string password)
+        => await InsertUserAsync(user, password, Array.Empty<Claim>(), null );
+
+    public async Task<IdentityResult> InsertUserAsync(User user, string password, Persoon? persoon)
+        => await InsertUserAsync(user, password, Array.Empty<Claim>(), persoon);
+
+    public async Task<IdentityResult> InsertUserAsync(User user, string password, Claim[] claims)
+        => await InsertUserAsync(user, password, claims, null);
+
+
+    public async Task<IdentityResult> InsertUserAsync(User user, string password, Claim[] claims, Persoon? persoon)
+    {
+        var result = await this._userManager.CreateAsync(user, password);
+        if (!result.Succeeded)
+        {
+            return result;
+        }
+        else
+        {
+            foreach (var claim in claims)
+            {
+                if (!IsValidClaim(claim))
+                {
+                    throw new InvalidOperationException($"The claim {claim.Type} with value {claim.Value} and value type {claim.ValueType} can not be added.");
+                }
+                result = await this._userManager.AddClaimAsync(user, claim);
+                if (!result.Succeeded)
+                {
+                    return result;
+                }
```

Comment:
Ik zou eerder hier throw-en. Dat zo'n AddClaim faalt is behoorlijk exceptioneel.